src/streaming/controllers/CmcdController.js

```diff
@@ -0,0 +1,733 @@
+
+        if (invalidRequests.length === enabledRequests.length) {
+            logger.error(`None of the request types are supported.`);
+            return false;
+        }
+
+        invalidRequests.map((k) => {
+            logger.warn(`request type ${k} is not supported.`);
+        });
+
+        return true;
+    }
+
+    function _checkTargetAvailableKeys(targetSettings) {
+        const cmcdAvailableKeysForMode = _getAvailableKeysForTarget(targetSettings);
+        const enabledCMCDKeys = targetSettings.enabledKeys;
+        
+        if (enabledCMCDKeys == null) {
+            return true;
+        }
+
+        if (enabledCMCDKeys.length === 0) {
+            return true;
+        }
+
+        const invalidKeys = enabledCMCDKeys.filter(k => !cmcdAvailableKeysForMode.includes(k));
+
+        if (invalidKeys.length === enabledCMCDKeys.length && enabledCMCDKeys.length > 0) {
+            const mode = targetSettings.cmcdMode;
+            if (!_loggedCmcdModeErrors.has(mode)) {
+                logger.error(`None of the keys are implemented for CMCD version 2 for mode ${targetSettings.cmcdMode}.`);
+                _loggedCmcdModeErrors.add(mode);
+            }
+            return false;
+        }
+        invalidKeys.map((k) => {
+            if (!_loggedCmcdModeErrors.has(k)) {
+                logger.warn(`key parameter ${k} is not implemented for CMCD version 2 for mode ${targetSettings.cmcdMode}.`);
+                _loggedCmcdModeErrors.add(k);
+            }
+        });
+
+        return true;
+    }
+
+    function _getAvailableKeysForTarget(targetSettings) {
+        const CMCD_V2_COMMON_KEYS = Constants.CMCD_V2_COMMON_AVAILABLE_KEYS;
+        const CMCD_V2_EVENT_MODE_KEYS = Constants.CMCD_V2_EVENT_MODE_AVAILABLE_KEYS;
+        const CMCD_V2_REQUEST_MODE_KEYS = Constants.CMCD_V2_REQUEST_MODE_AVAILABLE_KEYS;
+        const CMCD_V2_RESPONSE_MODE_KEYS = Constants.CMCD_V2_RESPONSE_MODE_AVAILABLE_KEYS;    
+        
+        var cmcdAvailableKeysForMode = [];
+
+        switch (targetSettings.cmcdMode) {
+            case Constants.CMCD_MODE.RESPONSE:
+                cmcdAvailableKeysForMode = CMCD_V2_COMMON_KEYS.concat(CMCD_V2_RESPONSE_MODE_KEYS);
+                break;
+            case Constants.CMCD_MODE.REQUEST:
+                cmcdAvailableKeysForMode = CMCD_V2_COMMON_KEYS.concat(CMCD_V2_REQUEST_MODE_KEYS);
+                break;
+            case Constants.CMCD_MODE.EVENT:
+                cmcdAvailableKeysForMode = CMCD_V2_COMMON_KEYS.concat(CMCD_V2_EVENT_MODE_KEYS);
+                break;
+        }
+
+        return cmcdAvailableKeysForMode;
+    }
+
+    function _createCmcdV2HeadersCustomMap() {
+        const cmcdVersion = settings.get().streaming.cmcd.version;
+        return cmcdVersion === 1 ? {} : { 
+            customHeaderMap: { 
+                [CmcdHeaderField.REQUEST]: ['ltc'],
+                [CmcdHeaderField.SESSION]: ['msd']
+            }
+        };
+    }
```

Comment:
The header mapping is CML was updated with the new CMCD v2 keys in version 0.15.1. This function should no longer be needed.